commands/provider/service_deployer.go

```diff
@@ -73,8 +75,27 @@ func (p *ServiceProvider) ServiceDeploy(path string, env map[string]string, stat
 }
 
 func deployServiceSendServiceContext(path string, env map[string]string, stream coreapi.Core_DeployServiceClient) error {
+	contextDir, relDockerfile, err := build.GetContextFromLocalDir(path, build.DefaultDockerfileName)
+	if err != nil {
+		return err
+	}
+
+	excludes, err := build.ReadDockerignore(contextDir)
+	if err != nil {
+		return err
+	}
+
+	if err := build.ValidateContextDirectory(contextDir, excludes); err != nil {
+		return fmt.Errorf("error checking context: %s", err)
+	}
+
+	// And canonicalize dockerfile name to a platform-independent one
+	relDockerfile = archive.CanonicalTarNameForPath(relDockerfile)
+	excludes = build.TrimBuildFilesFromExcludes(excludes, relDockerfile, false)
+
 	archive, err := archive.TarWithOptions(path, &archive.TarOptions{
```

Comment:
What about `xarchive` package?

Should we refacto most of the content of `deployServiceSendServiceContext` or simply delete `xarchive` package?

Reply:
Some other packages like service (or container) can use it, but it needs refactoring first. And here is small PR to add this to dev/deploy command